Assets/MRTK/Services/InputSystem/FocusProvider.cs

```diff
@@ -1226,40 +1226,48 @@ private static void QueryScene(IMixedRealityPointer pointer, IMixedRealityRaycas
                                 Array.Resize<Collider>(ref colliders, maxQuerySceneResults);
                             }
 
-                            int numColliders = UnityEngine.Physics.OverlapSphereNonAlloc(pointer.Rays[i].Origin, pointer.SphereCastRadius, colliders, ~UnityEngine.Physics.IgnoreRaycastLayer);
-
-                            if (numColliders > 0)
+                            Vector3 testPoint = pointer.Rays[i].Origin;
+                            Vector3 objectHitPoint = testPoint;
+                            GameObject closest = null;
+                            float closestDistance = Mathf.Infinity;
+                            int totalNumColliders = 0;
+
+                            // Go through each layerMask and ensure perform the appropriate OverlapSphereCalculation
+                            // Since this is usually done when a pointer passes a IsInteractionEnabled, maybe we can cache the selected colliders inside the pointer?
```

Comment:
Great suggestion! It would be good to file an issue and reference it here so we don't forget to do it in the future.

Reply:
Being tracked by #7629 now